common/src/main/java/tachyon/underfs/UnderFileSystem.java

```diff
@@ -364,6 +368,53 @@ public abstract OutputStream create(String path, short replication, int blockSiz
   public abstract String[] list(String path) throws IOException;
 
   /**
+   * Returns an array of strings naming the files and directories in the directory denoted by this
+   * abstract pathname, and all of its subdirectories.
+   *
+   * <p>
+   * If this abstract pathname does not denote a directory, then this method returns {@code null}.
+   * Otherwise an array of strings is returned, one for each file or directory in the directory and
+   * its subdirectories. Names denoting the directory itself and the directory's parent directory
+   * are not included in the result. Each string is a path relative to the given directory.
+   *
+   * <p>
+   * There is no guarantee that the name strings in the resulting array will appear in any specific
+   * order; they are not, in particular, guaranteed to appear in alphabetical order.
+   *
+   * @param path the path to list
```

Comment:
Since the rest of this doc calls it an "abstract pathname", should this be "the abstract pathname to list"?